### DISCUSSION TITLE
argocd app perms

### DISCUSSION THREAD

**Discussion by rwong2888** [argoproj/argo-cd#7196] [General]
Trying to implement argocd-sync-and-wait from https://argoproj-labs.github.io/argo-workflows-catalog/
What permissions would I need for that w/ a machine account?
```
argocd app sync {{inputs.parameters.application-name}} --revision {{inputs.parameters.revision}} {{inputs.parameters.flags}}
argocd app wait {{inputs.parameters.application-name}} --health {{inputs.parameters.flags}}
```

**Comment by wtam2018** [comment]
I am not sure what kind of machine account permission in your question.   If you are asking about argocd user account permission, I hope the follow help.   argocd-sync-and-wait requires some form of credentials which can be either argocd username/password or token.   The credentials will have permissions to run these commands (argocd app sync and argocd app wait) on your applications.   Please see here for more information about  using tokens https://argoproj.github.io/argo-cd/user-guide/projects/

**Reply by rwong2888** [reply]
What kind of permissions does that user need @wtam2018 ? Does it need admin or can it be lesser type of rbac?
Looking to create the necessary policy.csv for this
https://argoproj.github.io/argo-cd/operator-manual/rbac/#tying-it-all-together

**Reply by wtam2018** [reply]
@rwong2888  You shouldn't need admin.   
https://argoproj.github.io/argo-cd/operator-manual/rbac/#rbac-resources-and-actions
It looks like you just need `sync` and `get` on `applications`.
